.github/workflows/pretest.yml

```
@@ -36,6 +36,12 @@ jobs:
       run: |
         pre-commit run -a --show-diff-on-failure
 
+    - name: Typing
+      run: |
+        pip install numpy==1.25.* scipy==1.10.*
```

* Let's start with this combination and later expand the variation 😄 
* Maybe better to pin mypy version as well?
* Will `types-setuptools` be required?

```suggestion
        # TODO: Test against various NumPy/SciPy versions.
        pip install numpy==1.25.* scipy==1.10.*
```